### ISSUE TITLE
RBAC for UDP

### ISSUE THREAD

**Issue by gophobic** [envoyproxy/envoy#11656]
Is this something on your roadmap? I think it would be a good idea in order to be consistent having RBAC for all kinds of traffic at L4.

**Comment by lizan** [comment]
cc @yangminzhu

**Comment by yangminzhu** [comment]
This looks interesting, I think there are less information that could be used in RBAC for UDP request, probably just port, IP and maybe the payload. Do you have any thoughts of how the RBAC will be used with UDP?

Also curious does Envoy already support any other filters for UDP or is RBAC to be the first one?

**Comment by gophobic** [comment]
@lizan, @yangminzhu thanks. isn't the RBAC proto message used for both TCP and HTTP filters? you can reuse it for the UDP listener and just ignore rule items that don't relate to UDP like `header` and `url_path` which is what TCP does. I am just guessing here because I never really delved into Envoy's code.

**Comment by yangminzhu** [comment]
Yes we could use the same RBAC proto message and just ignore the inapplicable fields, I'm just curious if you have any concrete use cases of using RBAC with UDP.

**Comment by gophobic** [comment]
I get that UDP has far far less applications than TCP. But an obvious case is the UDP listener can allow or disallow requests to an upstream DNS server based on the source ip ranges.

**Comment by mattklein123** [comment]
Yes I think we can do this for UDP. One thing to keep in mind is that the current UDP filter stack is not complete and only allows a single filter currently. Someone would need to finish the UDP filter manager work so that we can support multiple filters.
